pkg/datarepair/checker/checker.go

```diff
@@ -72,13 +75,34 @@ func NewChecker(metainfo *metainfo.Service, repairQueue queue.RepairQueue, overl
 func (checker *Checker) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	return checker.Loop.Run(ctx, func(ctx context.Context) error {
-		err := checker.IdentifyInjuredSegments(ctx)
+	c := make(chan error)
+
+	go func() {
```

Comment:
rather than the go func()'s, channels, and loggers, I think we should use an errgroup here, similar to https://github.com/storj/storj/blob/master/satellite/peer.go#L551

Reply:
I would like to take this up in another PR.